src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -137,6 +151,62 @@ void AndroidDeviceControllerWrapper::OnPairingDeleted(CHIP_ERROR error)
     CallJavaMethod("onPairingDeleted", static_cast<jint>(error));
 }
 
+CHIP_ERROR AndroidDeviceControllerWrapper::GenerateNodeOperationalCertificate(const chip::PeerId & peerId, const chip::ByteSpan & csr,
+                                                                                int64_t serialNumber, uint8_t * certBuf,
+                                                                                uint32_t certBufSize, uint32_t & outCertLen)
+{
+    jmethodID method;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    err = FindMethod(GetEnvForCurrentThread(), mJavaObjectRef, "onOpCSRGenerationComplete", "([B)V", &method);
+    if (err != CHIP_NO_ERROR)
+   {
+      ChipLogError(Controller, "Error invoking onOpCSRGenerationComplete: %d", err);
+      return err;
+   }
+
+    // Initializing the KeyPair.
+     Initialize();
+
+    chip::X509CertRequestParams request = { serialNumber, mIssuerId,         mNow, mNow + mValidity, true, peerId.GetFabricId(),
+                                          true,         peerId.GetNodeId() };
+
+    chip::P256PublicKey pubkey;
+    ReturnErrorOnFailure(VerifyCertificateSigningRequest(csr.data(), csr.size(), pubkey));
+
+    ChipLogProgress(chipTool, "VerifyCertificateSigningRequest");
+
+
+    CHIP_ERROR generateCert = NewNodeOperationalX509Cert(request, chip::CertificateIssuerLevel::kIssuerIsRootCA, pubkey, mIssuer, certBuf, certBufSize,
+                                         outCertLen);
+    jbyteArray argument;
+    GetEnvForCurrentThread()->ExceptionClear();
+    N2J_ByteArray(GetEnvForCurrentThread(), csr.data(),csr.size(),argument);
+    GetEnvForCurrentThread()->CallVoidMethod(mJavaObjectRef, method, argument);
+    return generateCert;
+}
+
+CHIP_ERROR AndroidDeviceControllerWrapper::Initialize()
+{
+    chip::Crypto::P256SerializedKeypair serializedKey;
+    uint16_t keySize = static_cast<uint16_t>(sizeof(serializedKey));
+
+    if (SyncGetKeyValue(kOperationalCredentialsIssuerKeypairStorage, &serializedKey, keySize) != CHIP_NO_ERROR)
```

Comment:
Done

Reply:
I believe this should be compared with very specific CHIP_ERROR code, which indicate that specified item is not found in the keystore. For any other error the function should return that error instead of trying to generate a new key pair.